job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -184,13 +184,11 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
             newExecution.addListener(new GraphL());
             completed = new AtomicBoolean();
             logsToCopy = new LinkedHashMap<String,Long>();
-            newExecution.start();
-            
-            // It's only okay to initialise the global once the new FlowExecution
-            // has successfully started.
             execution = newExecution;
+            newExecution.start();
```

Comment:
@daniel-beck fair enough, I missed that. Thanks. I still think it should not be iniitalized until after it is successfully started. If it needs to be initialized before the call to `start`, then that suggests to me that there's a bigger problem somewhere else.